packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -418,14 +430,28 @@ class Slider extends React.Component {
       [classes.vertical]: vertical,
     });
 
-    const thumbClasses = classNames(classes.thumb, commonClasses);
-
     const trackProperty = vertical ? 'height' : 'width';
     const thumbProperty = vertical ? 'top' : 'left';
     const inlineTrackBeforeStyles = { [trackProperty]: this.calculateTrackBeforeStyles(percent) };
     const inlineTrackAfterStyles = { [trackProperty]: this.calculateTrackAfterStyles(percent) };
     const inlineThumbStyles = { [thumbProperty]: `${percent}%` };
 
+    /** Start Thumbnail Icon Logic Here */
+    const withIcon = !!Thumb;
+    const Thumbnail = withIcon
+      ? React.cloneElement(Thumb, {
+          ...Thumb.props,
+          className: `${classes.thumbIcon} ${Thumb.props.className || ''}`,
+        })
+      : null;
+    /** End Thumbnail Icon Logic Here */
+
+    const thumbClasses = classNames(
+      `${withIcon ? '' : classes.thumb}`,
+      commonClasses,
+      `${withIcon ? classes.thumbTransparent : ''}`,
+    );
```

Comment:
> You can leverage classNames implementation here by passing an object with the classnames as keys and a truthy value.

Done, I have made the changes. :)

> Why not apply classes.thumb always and classes.thumbIcon if an icon is passed and style accordingly. This would make more sense in my opinion. thumbTransparent just feels wrong. I'll always try to avoid naming my css classes after their look.

Regarding why I am not using `thumbIcon` is because I am using it for the `Thumbnail` itself, so instead of `thumbTransparent`  I used `thumbIconWrapper`

Reply:
Also I can not use the `classes.thumb` if there is a custom icon, because the recent issue you posted on active hovering state was because of this class.

Reply:
How would it look if you overwrite the conflicting styles? It might be confusing to users that this class is not always applied to the thumb.

Reply:
I have enabled `classes.thumb` even when it is used with a thumb icon.

Reply:
Also I just wanted to say thank you for taking the time out of your busy schedule, to review my code. This really means a lot. Thank you.